assets/js/custom/UserNotifications.js

```diff
@@ -0,0 +1,152 @@
+      document.getElementById('reaction-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('reaction-notif').style.color = 'rgba(0,0,0,0.87)'
+      document.getElementById('remix-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('remix-notif').style.color = 'rgba(0,0,0,0.87)'
+      all = false
+      follower = false
+      reactions = false
+      remixes = false
+    }
+  })
+  $(document).on('click', '#reaction-notif', function () {
+    if (reactions === false) {
+      $(this).css('background-color', '#acacac')
+      $(this).css('color', 'rgba(0,0,0,0.87)')
+      reactions = true
+      document.getElementById('all-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('all-notif').style.color = 'rgba(0,0,0,0.87)'
+      document.getElementById('follow-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('follow-notif').style.color = 'rgba(0,0,0,0.87)'
+      document.getElementById('comment-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('comment-notif').style.color = 'rgba(0,0,0,0.87)'
+      document.getElementById('remix-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('remix-notif').style.color = 'rgba(0,0,0,0.87)'
+      all = false
+      follower = false
+      comment = false
+      remixes = false
+    }
+  })
+  $(document).on('click', '#remix-notif', function () {
+    if (remixes === false) {
+      $(this).css('background-color', '#acacac')
+      $(this).css('color', 'rgba(0,0,0,0.87)')
+      remixes = true
+      document.getElementById('all-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('all-notif').style.color = 'rgba(0,0,0,0.87)'
+      document.getElementById('follow-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('follow-notif').style.color = 'rgba(0,0,0,0.87)'
+      document.getElementById('comment-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('comment-notif').style.color = 'rgba(0,0,0,0.87)'
+      document.getElementById('reaction-notif').style.backgroundColor = '#e0e0e0'
+      document.getElementById('reaction-notif').style.color = 'rgba(0,0,0,0.87)'
+      all = false
+      follower = false
+      comment = false
+      reactions = false
+    }
+  })
+  self.markAsRead = function (notificationId, type, seen, id) {
+    if (!seen) {
+      $.ajax({
+        url: self.markAsReadUrl + '/' + notificationId,
+        type: 'get',
+        success: function (data) {
+          if (data.success) {
+            self.updateBadgeNumber()
+            reloadResources(notificationId)
+          } else {
+            Swal.fire(somethingWentWrongError, notificationsClearError, 'error')
+          }
+        },
+        error: function () {
+          Swal.fire(somethingWentWrongError, notificationsClearError, 'error')
+        }
+      })
+    }
+    if (type === 'follow') {
```

Comment:
this type checks in order to redirect to a page have nothing to do with markAsRead

implicit action highly reduce the code readability and increase the cost to maintain a project over time. Please use an explicit action to handle those redirects